balancer-js/src/modules/pools/pool-types/concerns/stable/join.concern.integration.spec.ts

```diff
@@ -40,185 +39,93 @@ const slots = [
+    pool = await testPoolHelper.getPool();
+    const balances = pool.tokens.map((token) =>
       parseFixed(initialBalance, token.decimals).toString()
     );
     await forkSetup(
       signer,
-      tokensIn.map((t) => t.address),
+      pool.tokensList,
       slots,
       balances,
       jsonRpcUrl as string,
       14717479 // holds the same state as the static repository
     );
     signerAddress = await signer.getAddress();
   });
-
-  context('join transaction - join with encoded data', () => {
-    before(async function () {
-      this.timeout(20000);
-      amountsIn = tokensIn.map((t) =>
-        parseFixed(t.balance, t.decimals).div(amountsInDiv).toString()
-      );
-
-      [bptBalanceBefore, ...tokensBalanceBefore] = await getBalances(
-        [pool.address, ...pool.tokensList],
+  it('should join with encoded data', async () => {
+    const tokensIn = pool.tokensList;
+    const amountsIn = pool.tokens.map(({ decimals }, i) =>
+      parseFixed((i * 100).toString(), decimals).toString()
+    );
+    const slippage = '1';
+    const { to, data, minBPTOut, expectedBPTOut } = pool.buildJoin(
+      signerAddress,
+      tokensIn,
+      amountsIn,
+      slippage
+    );
+    const { transactionReceipt, balanceDeltas } =
+      await sendTransactionGetBalances(
+        tokensIn,
         signer,
-        signerAddress
-      );
-
-      const slippage = '1';
-
-      const { to, data, minBPTOut } = controller.buildJoin(
         signerAddress,
-        tokensIn.map((t) => t.address),
-        amountsIn,
-        slippage
-      );
-
-      const tx = { to, data };
-
-      bptMinBalanceIncrease = BigNumber.from(minBPTOut);
-      transactionReceipt = await (await signer.sendTransaction(tx)).wait();
-      [bptBalanceAfter, ...tokensBalanceAfter] = await getBalances(
-        [pool.address, ...pool.tokensList],
-        signer,
-        signerAddress
-      );
-    });
-
-    it('should work', async () => {
-      expect(transactionReceipt.status).to.eql(1);
-    });
-
-    it('price impact calculation', async () => {
-      const minBPTOut = bptMinBalanceIncrease.toString();
-      const priceImpact = await controller.calcPriceImpact(
-        amountsIn,
-        minBPTOut,
-        true
+        to,
+        data
       );
-      expect(priceImpact).to.eql('100000000006037');
-    });
-
-    it('should increase BPT balance', async () => {
-      expect(bptBalanceAfter.sub(bptBalanceBefore).gte(bptMinBalanceIncrease))
-        .to.be.true;
-    });
-
-    it('should decrease tokens balance', async () => {
-      for (let i = 0; i < tokensIn.length; i++) {
-        expect(
-          tokensBalanceBefore[i].sub(tokensBalanceAfter[i]).toString()
-        ).to.equal(amountsIn[i]);
-      }
-    });
+    const expectedMinBpt = subSlippage(
+      BigNumber.from(expectedBPTOut),
+      BigNumber.from(slippage)
+    ).toString();
+    expect(transactionReceipt.status).to.eq(1);
+    expect(BigInt(expectedBPTOut) > 0).to.be.true;
```

Comment:
We don't have a test checking for actual BPT balance increase. Should be able to add BPT address to `tokensForBalanceCheck` param for `sendTransactionGetBalances` function.

Reply:
@lgahdl - this still hasn't been updated as far as I can see. Take a look at the exit concern where we do:
```
          await sendTransactionGetBalances(
            [pool.address, ...pool.tokensList],
            signer,
            signerAddress,
            to,
            data
          );
```
to get the BPT delta.